rclcpp/src/rclcpp/duration.cpp

```diff
@@ -47,17 +47,14 @@ Duration::Duration(std::chrono::nanoseconds nanoseconds)
   rcl_duration_.nanoseconds = nanoseconds.count();
 }
 
-Duration::Duration(const Duration & rhs)
-{
-  rcl_duration_.nanoseconds = rhs.rcl_duration_.nanoseconds;
-}
+Duration::Duration(const Duration & rhs) = default;
 
 Duration::Duration(
   const builtin_interfaces::msg::Duration & duration_msg)
 {
   rcl_duration_.nanoseconds =
-    static_cast<rcl_duration_value_t>(RCL_S_TO_NS(duration_msg.sec));
-  rcl_duration_.nanoseconds += duration_msg.nanosec;
+    RCL_S_TO_NS(static_cast<rcl_duration_value_t>(duration_msg.sec));
```

Comment:
why changing the order here?
https://github.com/ros2/rcutils/blob/e7eb74c7a11cf207aa9330d77f44098339528cac/include/rcutils/time.h#L30
https://github.com/ros2/rcutils/blob/e7eb74c7a11cf207aa9330d77f44098339528cac/include/rcutils/time.h#L48

Reply:
For the result it should not matter in this case, because the result of `RCL_S_TO_NS(duration_msg.sec)` has type `int64_t` implicitly and the `static_cast<rcl_duration_value_t>` is a no-op.

It was mainly for consistency with
https://github.com/ros2/rclcpp/blob/683ac64ee4edf52ae3a4cd1a722c17c4bd81b562/rclcpp/src/rclcpp/duration.cpp#L84-L90

So this change addresses the
> All other changes are mainly cosmetic, either to make implicit casts between signed and unsigned integer types explicit, or to match the implementation of constructors and respective assignment operators.

part of the description.